includes/helper/class-wp-job-manager-helper.php

```diff
@@ -449,39 +531,55 @@ public function extra_headers( $headers ) {
 	/**
 	 * Returns list of installed WPJM plugins with managed licenses indexed by product ID.
 	 *
-	 * @param bool $active_only Only return active plugins.
+	 * @since $$next-version$$ Added required $keyed_by_filename parameter.
+	 *
+	 * @param bool $active_only       Only return active plugins.
+	 * @param bool $keyed_by_filename Key by plugin filename instead of product slug. Allows for multiple plugins with the same product slug.
 	 * @return array
 	 */
-	public function get_installed_plugins( $active_only = true ) {
+	public function get_installed_plugins( $active_only = true, $keyed_by_filename = null ) {
 		if ( ! function_exists( 'get_plugins' ) ) {
 			require_once ABSPATH . 'wp-admin/includes/plugin.php';
 		}
 
+		if ( null === $keyed_by_filename ) {
+			_doing_it_wrong( __METHOD__, 'The $keyed_by_filename parameter is required.', '$$next-version$$' );
+			$keyed_by_filename = false;
+		}
+
+		$wpjm_plugins = [];
+		$plugins      = get_plugins();
+
 		/**
-		 * Clear the plugin cache on first request for installed WPJM add-on plugins.
+		 * Clear the plugin cache on first request for installed WPJM add-on plugins when no plugins found.
 		 *
 		 * @since 1.29.1
+		 * @since $$next-version$$ Only do this when we don't see WP Job Manager in the plugins list.
 		 *
 		 * @param bool $clear_plugin_cache True if we should clear the plugin cache.
 		 */
-		if ( ! self::$cleared_plugin_cache && apply_filters( 'job_manager_clear_plugin_cache', true ) ) {
+		if ( ! self::$cleared_plugin_cache && apply_filters( 'job_manager_clear_plugin_cache', ! isset( $plugins[ JOB_MANAGER_PLUGIN_BASENAME ] ) ) ) {
 			// Reset the plugin cache on the first call. Some plugins prematurely hydrate the cache.
 			wp_clean_plugins_cache( false );
 			self::$cleared_plugin_cache = true;
+			$plugins                    = get_plugins();
 		}
```

Comment:
We were being too eager on this before for an edge case. Some plugins were polluting the cache. I think it is (a) Rare and (b) Sniffable by seeing if the (obviously active) WP Job Manager plugin is listed. Before we were flushing the plugin update cache _way too often._